### PR TITLE
elrond-wasm 0.6.0

### PR DESCRIPTION
**Pull request by andrei-marinica**
Contains:
- elrond-wasm 0.6.0
- example contracts
- test contracts
- standard module: feature module

### CODE REVIEW

File: examples/factorial/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+#[elrond_wasm_derive::contract(FactorialImpl)]
+pub trait Factorial {
+
+    #[init]
+    fn init(&self) {
+    }
+
+    #[endpoint]
+    fn factorial(&self, value: BigUint) -> BigUint {
```
Comment (camilbancioiu) [inline examples/factorial/src/lib.rs:14]: :+1: for `BigUint`

**Review by camilbancioiu — APPROVED** [review]

File: build-wasm.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+### EXAMPLES ###
+
+cd examples/adder/wasm
+RUSTFLAGS='-C link-arg=-s' \
+cargo build --target=wasm32-unknown-unknown --release
+cd ..
+mkdir -p output
+cp wasm/target/wasm32-unknown-unknown/release/adder_wasm.wasm output/adder.wasm
+cd ../..
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply build-wasm.sh:13]: can do, but only after erdpy is updated

File: examples/adder/wasm/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+opt-level = "z"
+lto = true
+debug = false
+panic = "abort"
+
+[dependencies]
+adder = { path = ".." }
+elrond-wasm-output = { path = "../../../elrond-wasm-output", features=["wasm-output-mode"]}
+
+[workspace]
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply examples/adder/wasm/Cargo.toml:21]: Yes, absolutely!! The only way this works!

File: elrond-wasm-derive/src/contract_macro_main.rs [hunk 4]
```diff
@@ -13,29 +13,33 @@ pub fn process_contract(
...
+
+    proc_macro::TokenStream::from(quote! {
+        #[macro_use]
+        extern crate elrond_wasm;
+
+        #contract_impl
+
+        #wasm_callback_endpoint
+    })
 }
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply elrond-wasm-derive/src/contract_macro_main.rs:45]: ?

File: examples/adder/.gitignore [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Generated by Cargo
+# will have compiled files and executables
+/target/
+*/target/
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply examples/adder/.gitignore:4]: Well, it doesn't suffice on my machine

**Pull request merged**